gnpy/core/elements.py

```diff
@@ -263,62 +242,49 @@ def to_json(self):
                 'type_variety'  : self.type_variety,
                 'params'        : {
                 #have to specify each because namedtupple cannot be updated :(
-                    'type_variety'  : self.type_variety,
-                    'length'        : self.length/UNITS[self.params.length_units],
-                    'loss_coef'     : self.loss_coef*1e3,
-                    'length_units'  : self.params.length_units,
-                    'att_in'        : self.att_in,
-                    'con_in'        : self.con_in,
-                    'con_out'       : self.con_out
+                    'length'        : round(self.params.length * 1e-3, 6),
```

Comment:
This removes
`self.length/UNITS[self.params.length_units]`
 Does this means that lengths are in km and unit is no more a relevant attribute in json ? then README should be corrected

Reply:
The length unit is still useful in the configuration .json because the user can express the length in any unit he/she prefers using this parameter. No matter what their choice is, the length of the fiber instance is converted into meters as a fundamental unit, meaning that the length_unit fiber instance parameter always end up as 'm'.
In the quoted piece of code, a json representative of the fiber is created. We think that expressing length in km is a reasonable choice to support legibility.

Reply:
OK, I understand, however if I use an input json network file with units in m, then transmission_main does not work anymore:
OverflowError: (34, 'Numerical result out of range')
 I think that the initialization of Fiber in
`network_from_json(json_data, equipment)`
leaves the possibility to have other units, so that 
`self.params.length` can have different units, and maybe this gives issues in the initialization of params ?

Reply:
The point here is that length_unit should concern all quantities in fiber parameters that include a unit of length. This is also the case for loss_coef; if 'm' is chosen as length_unit, loss_coeff should be expressed in dB/m in the configuration .json.

Reply:
This happens because unit conversion is applied also to `loss_coef` while in the existing code it is assumed that `loss_coef` is always specified in dB/km.

Reply:
> The point here is that length_unit should concern all quantities in fiber parameters that include a unit of length.

But you don't apply conversion to dispersion and gamma which also include a unit of length.

Reply:
> > The point here is that length_unit should concern all quantities in fiber parameters that include a unit of length.
> 
> But you don't apply conversion to dispersion and gamma which also include a unit of length.

This is true. Nevertheless, since both dispersion and gamma are already expressed in fundamental units, we thought that this would be always the case as a requirement.

Reply:
Right now, the user has the flexibility to use units in the topo file for the length only, independently from the units of the lineic loss which is assumed to be always in dB/km. I would prefer to keep this as it is in this PR (changing the `/ self._length_units_factor`  into a `/ UNITS['km']` in parameters.py, lines 169 to 175). The question on meter or km is trivial, but some user may have prepared all their files in meters and I would prefer to avoid a refactor change that also changes user experience. Would it be possible to postpone this discussion/(unit uniformisation) when we will discuss yang ?

Reply:
The choice of 'm' or 'km' involves some subtlety - in the current version's configuration json the lineic loss is always assumed to be in dB/km, however, we want the fiber parameter loss_coef to be expressed in fundamental units (dB/m). This means that it is not possible to avoid a duplication of this parameter in this case; we thought that using the length_units would solve this issue. In any case, in the last commit we restore this duality of loss_coef units (dB/km in the configuration files, dB/m in the fiber instance) in order to guarantee no changes for the user, but this introduces an ambiguity between the instance's value and the class when called as a dictionary (using asdict).